mksnapshot.sh

```diff
@@ -9,41 +9,43 @@
 
 echo "RUN: $0"
 if [ $# -lt 1 ]; then
-	echo "Usage: $0 [branch]"
-	echo "Where [branch] is the path under /bzr/ to the branch to snapshot."
+	echo "Usage: $0 [branch] [tag]"
+	echo "Where [branch] is the name of the branch to snapshot."
+	echo "Where [tag] is the name of the tag on that branch to snapshot."
 	exit 1
 fi
 # VCS details
-module=squid3
-BZRROOT=${BZRROOT:-/bzr}
-
-# generate a tarball name from the branch ($1) note that trunk is at
-# /bzr/trunk, but we call it 3.HEAD for consistency with CVS (squid 2.x), and
-# branches are in /bzr/branches/ but we don't want 'branches/' in the tarball
-# name so we strip that.
-branchpath=${1:-trunk}
-tag=${2:-`basename $branchpath`}
+if test "x${VCSROOT}" = "x"; then
+	echo "VCSROOT needs to be set to the base path for the repository to snapshot."
+	exit 1
+fi
+
+# generate a tarball name from the branch ($1) at tag ($2)
+branch=${1:-master}
+tag=${2:-HEAD}
 startdir=${PWD}
 date=`env TZ=GMT date +%Y%m%d`
 
-tmpdir=${TMPDIR:-${PWD}}/${module}-${tag}-mksnapshot
+tmpdir=${TMPDIR:-${PWD}}/squid-${tag}-mksnapshot/
 
 rm -rf ${tmpdir}
 trap "echo FAIL-BUILD_${VERSION} ; rm -rf ${tmpdir}" 0
+mkdir ${tmpdir}
 
 rm -f ${tag}.out
-bzr export ${tmpdir} ${BZRROOT}/${module}/${branchpath} || exit 1
+git checkout ${branch} || exit 1
+(git archive --format=tar ${tag} | tar -xC ${tmpdir}) || exit 1
```

Comment:
It sounds like the problems with archiving are stemming from using HEAD, which we should not use, and which is discussed elsewhere. Yes, using (unqualified) HEAD requires a checkout because there are many HEADs (one per branch).

> [git checkout] validates that we have a valid repository and the branch actually exists before the archive is attempted

While true, that statement seems like an argument _against_ doing `checkout` before `archive`! We do not actually care whether the branch existed some time before the archive. We care that it existed for the archive command to succeed.

This is not a huge deal, but if we can avoid checkout, we should.